### DISCUSSION TITLE
Country name after the name of diocese

### DISCUSSION THREAD

**Discussion by tukusejssirs** [romcal/romcal#318] [Ideas]
@emagnier, I’ve just found out that there are two dioceses of Saint-Denis:

- [Diocese of Saint-Denis in France](https://en.wikipedia.org/wiki/Roman_Catholic_Diocese_of_Saint-Denis):
  - in French: _Diocèse de Saint-Denis en France_;
  - in Latin: _Diœcesis Sancti Dionysii in Francia_;
  - see also: https://saint-denis.catholique.fr/
- [Diocese of Saint-Denis de La Réunion](https://en.wikipedia.org/wiki/Roman_Catholic_Diocese_of_Saint-Denis_de_La_R%C3%A9union):
  - in French: _Diocèse de Saint-Denis de La Réunion_;
  - in Latin: _Diœcesis Sancti Dionysii Reunionis_.

I suggest to _always_ add the country of the diocese to the key name. I also suggest to add the country of the diocese in the key value **unless** the locale (set by romcal user) is used as an official language of the country where the particular calendar (set by romcal user) is used.

Reasoning:

- not every diocese is well-known world-wide, therefore in non-usual country-locale combinations, we should include `, {country}` (or locale equivalent);
- for key names, it might be the default to include `_{country}` in order to distinguish them.

---

Note that currently we cannot output two different key values, but this might change when we implement the patronage title generation (#???), because this could use similar approach, but still, we need to add another subobject into the l10n files (e.g. `dioceses`) in order to list all supported and/or otherwise used diocese names.

When agree on this, we need to update the key name generation rules.

---

I don’t actually know how to resolve is the country name of the Saint-Denis dioceses (esp in French):

- should we always use `en France` and `de La Réunion`?
- as both dioceses are geographically part of France (although Réunion is immediately subject to the Holy See and it’s a member of the Episcopal Conference of the Indian Ocean), we cannot use the suggested rule as it is written above.

For these two dioceses, I suggest to add additional rules:

- for the `Diocese of Saint-Denis, France`, always use either `, France` or `en France`;
- for the `Diocese of Saint-Denis, Réunion`, always use either `, Réunion` or `de La Réunion`.

---

For example:

- Diocese of Saint-Denis, France:
  - output the country name;
- Diocese of Saint-Denis, Réunion:
  - output the country name;
- Archdiocese of Paris, France:
  - no country in the key value when the country is France  **and** locale is `fr*`;
  - else output the country name;
- Archdiocese of Bratislava, Slovaka:
  - no country in the key value when the country is Slovakia (or whatever is the  **and** locale is `sk*`;
  - else output the country name.

**Comment by a420gmoon** [comment]
Hi 👋, I am Moon Das from India.

**Reply by tukusejssirs** [reply]
Hey, @a420gmoon, can I help you with something? :wink:
